In [1]:
# importing the necessary libraries for uploading data and data manupulations
import pandas as pd 
import numpy as np 

In [2]:
a0 = pd.Series([230,1,1,1,1,0,1])
a1 = pd.Series([60,1,1,1,1,0,1])
a2 = pd.Series([51,1,1,0,1,0,1])
a3 = pd.Series([432,0,1,1,0,1,1])
a4 = pd.Series([121,1,1,0,1,1,0])
a5 = pd.Series([89,0,0,1,0,1,0])

new = pd.Series([95,1,0,1,1,0,1])


In [3]:
# 1) Euclidean distance between the new customer and each of the other customers.
dist_newa0 = np.sqrt(np.sum([(x-y)**2 for x, y in zip(new, a0)]))     
dist_newa1 = np.sqrt(np.sum([(x-y)**2 for x, y in zip(new, a1)]))     
dist_newa2 = np.sqrt(np.sum([(x-y)**2 for x, y in zip(new, a2)])) 
dist_newa3 = np.sqrt(np.sum([(x-y)**2 for x, y in zip(new, a3)]))
dist_newa4 = np.sqrt(np.sum([(x-y)**2 for x, y in zip(new, a4)])) 
dist_newa5 = np.sqrt(np.sum([(x-y)**2 for x, y in zip(new, a5)])) 

In [4]:
print("The Euclidean distance from customer new to A is", dist_newa0)
  
print("The Euclidean distance from customer new to B is", dist_newa1)

The Euclidean distance from customer new to A is 135.00370365289984
The Euclidean distance from customer new to B is 35.014282800023196


In [5]:
print("The Euclidean distance from customer new to A is", dist_newa2)
  
print("The Euclidean distance from customer new to B is", dist_newa3)

The Euclidean distance from customer new to A is 44.02272140611028
The Euclidean distance from customer new to B is 337.0059346658453


In [6]:
print("The Euclidean distance from customer new to A is", dist_newa4)
  
print("The Euclidean distance from customer new to B is", dist_newa5)

The Euclidean distance from customer new to A is 26.076809620810597
The Euclidean distance from customer new to B is 6.324555320336759


The distance is minimum of  6.32 between the new custmer and 6th custmer.
As the custmer 6th has baby shower registry ( registry = 1 ), the new custmer should also have a baby shower registry 
as per minimum distance criteria.

In [7]:
# 2) Standardize (normalize) the predictor variables
df1 = pd.read_csv('Classfication.csv')
df1


Spent60Norm  Zinc10Norm  Zinc90Norm  Mag10Norm  Mag90Norm  Cotton10Norm  \
0     0.558009     0.58554     0.58554    0.58554    0.58554     -0.801784   
1    -0.690170     0.58554     0.58554    0.58554    0.58554     -0.801784   
2    -0.756250     0.58554     0.58554   -1.46385    0.58554     -0.801784   
3     2.041140    -1.46385     0.58554    0.58554   -1.46385      1.069045   
4    -0.242294     0.58554     0.58554   -1.46385    0.58554      1.069045   
5    -0.477245    -1.46385    -1.46385    0.58554   -1.46385      1.069045   
6    -0.433192     0.58554    -1.46385    0.58554    0.58554     -0.801784   
7          NaN         NaN         NaN        NaN        NaN           NaN   
8          NaN         NaN         NaN        NaN        NaN           NaN   
9     0.982480     0.00000     4.20000    0.00000    0.00000      0.000000   

   Cotton90Norm  
0       0.58554  
1       0.58554  
2       0.58554  
3       0.58554  
4      -1.46385  
5      -1.46385  
6       0.58554  
7           NaN  
8           NaN  
9       0.00000

In [8]:
a0Norm = pd.Series([0.558009,0.58554,0.58554,0.58554,0.58554,-0.801784,0.58554])
a1Norm = pd.Series([-0.690170,0.58554,0.58554,0.58554,0.58554,-0.801784,0.58554])
a2Norm = pd.Series([-0.756250,0.58554,0.58554,-1.46385,0.58554,-0.801784,0.58554])
a3Norm = pd.Series([2.041140,-1.46385,0.58554,0.58554,-1.46385,1.069045,0.58554])
a4Norm = pd.Series([-0.242294,0.58554,0.58554,-1.46385,0.58554,1.069045,-1.46385])
a5Norm = pd.Series([-0.477245,-1.46385,-1.46385,0.58554,-1.46385,1.069045,-1.46385])
a6Norm = pd.Series([-0.433192,0.58554,-1.46385,0.58554,0.58554,-0.801784,0.58554])

In [9]:
dist_newa0N = np.sqrt(np.sum([(x-y)**2 for x, y in zip(a6Norm, a0Norm)]))     
dist_newa1N = np.sqrt(np.sum([(x-y)**2 for x, y in zip(a6Norm, a1Norm)]))     
dist_newa2N = np.sqrt(np.sum([(x-y)**2 for x, y in zip(a6Norm, a2Norm)])) 
dist_newa3N = np.sqrt(np.sum([(x-y)**2 for x, y in zip(a6Norm, a3Norm)]))
dist_newa4N = np.sqrt(np.sum([(x-y)**2 for x, y in zip(a6Norm, a4Norm)])) 
dist_newa5N = np.sqrt(np.sum([(x-y)**2 for x, y in zip(a6Norm, a5Norm)])) 

In [10]:
print("The Euclidean distance from customer new to A is", dist_newa0N)
  
print("The Euclidean distance from customer new to B is", dist_newa1N)

The Euclidean distance from customer new to A is 2.276505830104768
The Euclidean distance from customer new to B is 2.0654387099558287


In [11]:
print("The Euclidean distance from customer new to A is", dist_newa2N)
  
print("The Euclidean distance from customer new to B is", dist_newa3N)

The Euclidean distance from customer new to A is 2.916224479624982
The Euclidean distance from customer new to B is 4.714055378309105


In [12]:
print("The Euclidean distance from customer new to A is", dist_newa4N)
  
print("The Euclidean distance from customer new to B is", dist_newa5N)

The Euclidean distance from customer new to A is 4.017018958126163
The Euclidean distance from customer new to B is 4.012722259308511


When predictor variables are standardized the distance is minimum of 2.07 is between 2nd and new custmer.
Custmer second is predicted as no registry = 0 , therefore this time new custmer will also be predicted as no registry = 0.
By making the predictors standardized the result has been changed.